### ISSUE TITLE
Missing typehints are not detected

### ISSUE THREAD

**Issue by OndraM** [lmc-eu/php-coding-standard#19]
It seems phpdoc_add_missing_param_annotation (aka PhpdocAddMissingParamAnnotationFixer) does not work as I expected - it only works if there is already at least something (even just comment) in 
phpdoc.

If phpdoc is missing, it won't say anything :-/.

This causes no error:
```php
    public function missingTypes($text, $date): void
    {
        // ...
    }
```

While this will:
```php
    /**
     * You should use types. Really.
     */
    public function missingTypes($text, $date): void
    {
        // ...
    }
```

And it will suggest this change:
```diff
     /**
      * You should use types. Really.
+     * @param mixed $text
+     * @param mixed $date
      */
     public function missingTypes($text, $date): void
     {
```

But this is expected behavior of the Fixer: https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2464

So we should find something to handle this properly.

**Comment by legendik** [comment]
How about this Slevomat sniff? https://github.com/slevomat/coding-standard#slevomatcodingstandardtypehintstypehintdeclaration-

**Comment by OndraM** [comment]
@legendik Nice siggestion, thanks! It detects this just fine. However, it also finds lots of false positives :-/. And for some reason removes other annotations - `@internal`, `@codeCoverageIgnore`... 

Also it seems kinda over-strict: it requires to specify internal types of array etc. (Which is nice, but from fast check on existing codebases, this may cause a *lot* violations). But we may at least consider using this in some kind of "strict" level rules... Also I will have a deeper look new  week(s).

**Comment by MortalFlesh** [comment]
Well about `specify internal types of array`IMHO it is nice on one side, but on the other.. is it really so better if you have `array<string>` instead of just `array` ?

Yes you know what you _should_ send there, but there is no garantee (maybe phpstan could check it, but I'm not sure how deep it is possible...).

If there are objects in array, it's great, because IDE know what to show you if you iterate through array items, but scalar types.. well I dont know really..

**Comment by OndraM** [comment]
Closing this, as it is reported by phpstan level 6 and above.

```
Method Lmc\Foo::missingTypes() has parameter $date with no typehint specified.  
```
